service/observe/subscription.go

```diff
@@ -40,6 +65,12 @@ type Subscriber struct {
 	TopicName string // contains the name that this subscriber is registered with
 }
 
+type promMetrics struct {
```

Comment:
This does not provide any value and the code can work without. Please remove and use package level variables without any wrapper.